backend/src/v1/services/external-consumer-service.ts

```diff
@@ -120,72 +144,131 @@ const externalConsumerService = {
+           left join pay_transparency.employee_count_range as employee_count_range on employee_count_range.employee_count_range_id = report.employee_count_range_id
+           where report_status = 'Published'
+               and (report.update_date >= ${convert(startDt).toDate()}
+                    and report.update_date < ${convert(endDt).toDate()})
+           union
+               (select report.report_id,
+                       report.report_history_id as report_change_id,
+                       report.company_id,
+                       report.user_id,
+                       report.user_comment,
+                       report.employee_count_range_id,
+                       report.naics_code,
+                       report.report_start_date,
+                       report.report_end_date,
+                       report.create_date,
+                       report.update_date,
+                       report.create_user,
+                       report.update_user,
+                       report.report_status,
+                       report.revision,
+                       report.data_constraints,
+                       report.is_unlocked,
+                       report.reporting_year,
+                       report.report_unlock_date,
+                       naics_code.naics_label,
+                       company.company_id,
+                       company.company_name,
+                       company.bceid_business_guid,
+                       company.address_line1,
+                       company.address_line2,
+                       company.city,
+                       company.province,
+                       company.country,
+                       company.postal_code,
+                       employee_count_range.employee_count_range
+                from pay_transparency.report_history as report
+				        left join pay_transparency.naics_code as naics_code on naics_code.naics_code = report.naics_code
+                left join pay_transparency.pay_transparency_company as company on company.company_id = report.company_id
+                left join pay_transparency.employee_count_range as employee_count_range on employee_count_range.employee_count_range_id = report.employee_count_range_id
+                where report_status = 'Published'
+                    and (report.update_date >= ${convert(startDt).toDate()}
+                         and report.update_date < ${convert(endDt).toDate()})))
+      order by update_date
+      offset ${offset}
+      limit ${limit}) as reports
+
+left join
+    (select data.report_id,
+            data.calculation_code_id,
+            data.value,
+            data.is_suppressed,
+            code.calculation_code
+     from
+         (select data.report_id,
+                 data.calculation_code_id,
+                 data.value,
+                 data.is_suppressed
+          from pay_transparency.pay_transparency_calculated_data as data
```

Comment:
I think you should add a "where update_date is between start and end" on the calculated data. otherwise the database is going to union the entirety of both tables. How close are the update_date columns of the calculated data table and report table? do we need to be concerned that the say the report table is updated before the "end" and the calculated date is updated after the "end"?